contk/metric/metric.py

```diff
@@ -78,8 +78,10 @@ def forward(self, data):
 				if not np.allclose(expsum, [1] * single_length):
 					raise ValueError("data[gen_prob_key] must be processed after log_softmax.")
 
-			self.word_loss += -np.sum(gen_prob[i][\
-				list(range(single_length-1)), resp[i][1:single_length]])
+			# self.word_loss += -np.sum(gen_prob[i][\
+			# 	list(range(single_length-1)), resp[i][1:single_length]])
+			for x, y in zip(list(range(single_length - 1)), resp[i][1:single_length]):
```

Comment:
This is a bottkeneck of performance. Try converting numpy.array to see which way will be faster. 
(By the way, you can use enumerate instead of zip(range(XX), XX))